### DISCUSSION TITLE
'input-remapper-control --list' without 'sudo'

### DISCUSSION THREAD

**Discussion by DreadPirateLynx** [sezanzeb/input-remapper#482] [Q&A]
When I run `input-remapper-control --list` on my system, I get the list of connected devices, but someone else I'm helping requires the use of `sudo` to get any information. What is causing this and how can we remove the `sudo` requirement for them?

**Comment by jonasBoss** [comment]
This is due to the permissions for the `/dev/input/eventXX` files, I think those are set by udev-rules and can differ from device to device.
On my system some devices can be read without root permissions while others can not. 

You could write your own udev-rules to update the permissions in my experience this can have interesting consequences:
I have a PS5 controller which has a gyroscope builtin. When I update the permissions the controller and its gyroscope would show up as 2 different joysticks on my system. This causes some games to get very confused.

**Comment by negora** [comment]
I'm sorry for resurrecting this thread, but I'm interested in loading presets without the need of entering my password every time (in other words: without doing `sudo`).

I'm using Input Remapper 2.0.1 in Debian 12.5. My `event*` devices have `rw` permissions for user `root` and group `input`. I've added my user to the `input` group, but I'm still getting an empty list when I run `input-remapper-control --list-devices` without `sudo`.

My plan is to automatically change the preset when a window is focused (thanks to KDE Plasma shortcuts and events). I doubt that the preset load is fast enough to make it usable, but I want to give it a try.

For now, I'm using an alternative method. I've remapped my mouse buttons (with Input Remapper) to some global keyboard shortcuts defined in Plasma. Then, these shortcuts are linked to a shell script that reads a parameter representing the key combination and, with the help of `xdotool`, I find out which window is active/focused, the application it belongs to (using the `WM_NAME` property of the window) and run the real shortcut that is specific to the application.

This is a simple diagram: Input Remapper -> Plasma shortcuts -> Shell script -> xdotool. It's working fast enough for my purposes, but I would love getting rid of the global shortcuts if possible.

Thank you.

**Reply by nbngtn** [reply]
omg how do you get the app to run on Debian? I'm running trixie with kde. Whenever I try to open up the app, it will just show a loading taskbar icon for a while, then disappear as if it closed itself. Been trying all sorts of way to make it work but cant :(

**Reply by negora** [reply]
@nbngtn I guess you're referring to the graphical application. Have you tried to run `input-remapper-gtk` from the command line and see what's the output?

Right now, `input-remapper-gtk` runs fine in my machine, but I believe I had a problem similar to yours months ago (although I use Bookworm, not Trixie). Maybe I solved it by adding my user to the group `input`, but I'm not sure...

**Comment by mercster** [comment]
Sorry to revive as well... on Fedora I have the same setup (group 'input' has rw access to /dev/input/event*)... so the GUI should be able to run without root permissions after I added my user to the group.  I want to try and find where to rip out the part of the program that prompts for root password, but I dunno where to start, python is weird.

EDIT: I do see my devices as my non-root user with --list.  Just need to get the gui to stop...
